packages/rocketchat-lib/server/lib/interceptDirectReplyEmails.js

```diff
@@ -0,0 +1,94 @@
+const Imap = require('imap');
+
+RocketChat.imapIntercepter = function() {
+	const imap = new Imap({
+		user: RocketChat.settings.get('IMAP_Username'),
+		password: RocketChat.settings.get('IMAP_Password'),
+		host: RocketChat.settings.get('IMAP_Host'),
+		port: RocketChat.settings.get('IMAP_Port'),
+		debug: RocketChat.settings.get('IMAP_Debug') ? console.log : false,
+		tls: !RocketChat.settings.get('IMAP_IgnoreTLS')
+	});
+
+	function openInbox(cb) {
+		imap.openBox('INBOX', false, cb);
+	}
+
+	function getEmails(imap) {
+		imap.search(['UNSEEN'], function(err, newEmails) {
+			if (err) {
+				console.log(err);
```

Comment:
Should you return after the console.log? If error exists `newEmails` will be empty probably and the next if will throw an exception.